File: packages/adapters/nexus-adapter-sdks/nexus-adapter-sdk-go/targets.go
package nexadapter

import "strings"

func RequireContainerTarget(target DeliveryTarget) (string, error) {
	containerID := strings.TrimSpace(target.Channel.ContainerID)
	if containerID == "" {
		return "", ErrTargetContainerRequired
	}
	return containerID, nil
}

func ReadThreadTarget(target DeliveryTarget) string {
	return strings.TrimSpace(target.Channel.ThreadID)
}

func ReadReplyToTarget(target DeliveryTarget) string {
	return strings.TrimSpace(target.ReplyToID)
}

File: packages/adapters/nexus-adapter-sdks/nexus-adapter-sdk-ts/src/targets.ts
import type { ChannelRef } from "./protocol.js";

function asNonEmptyString(value: string | undefined): string | undefined {
  const trimmed = value?.trim();
  return trimmed ? trimmed : undefined;
}

export function requireContainerTarget(channel: ChannelRef): string {
  const value = asNonEmptyString(channel.container_id);
  if (!value) {
    throw new Error("channel.container_id is required");
  }
  return value;
}

export function readThreadTarget(channel: ChannelRef): string | undefined {
  return asNonEmptyString(channel.thread_id);
}

export function readReplyToTarget(replyToID: string | undefined): string | undefined {
  return asNonEmptyString(replyToID);
}
